### PR TITLE
gpu logit processing for rank classification models

### PR DESCRIPTION
**Pull request by jagnusson**
Improved speed of rank classification models by aggregating sequence logits on GPU rather than on CPU

## Speed Up

Screen Shot 2022-08-19 at 10 26 00 AM

### CODE REVIEW

**Review by dirkgr — APPROVED** [review]
I find it shocking that such a small number of floats can still produce such a large performance difference.
